modules/lang-painless/src/main/java/org/elasticsearch/painless/action/PainlessExecuteAction.java

```diff
@@ -203,6 +215,34 @@ static ContextSetup parse(XContentParser parser, Void context) throws IOExceptio
                 query = in.readOptionalNamedWriteable(QueryBuilder.class);
             }
 
+            static Tuple<String, String> parseClusterAliasAndIndex(String indexExpression) {
+                if (indexExpression == null) {
+                    return new Tuple<>(null, null);
+                }
+                String trimmed = indexExpression.trim();
+                if (trimmed.startsWith(":") || trimmed.endsWith(":")) {
+                    throw new IllegalArgumentException(
+                        "Unable to parse one single valid index name from the provided index: [" + indexExpression + "]"
+                    );
+                }
+
+                // TODO: we are directly parsing and accepting user input here - any security sanitization/checking-of-input required here?
+                String[] parts = indexExpression.split(":", 2);
+                if (parts.length == 1) {
+                    return new Tuple<>(null, parts[0]);
+                } else if (parts.length == 2 && parts[1].contains(":") == false) {
+                    return new Tuple<>(parts[0], parts[1]);
+                } else {
+                    throw new IllegalArgumentException(
+                        "Unable to parse one single valid index name from the provided index: [" + indexExpression + "]"
+                    );
+                }
+            }
```

Comment:
would it make sense to reuse the parsing code from RemoteClusterService somehow?

Reply:
I tried that but it didn't work as I hoped. I'll go back and review it again and document it here.

Reply:
I thought so, thanks.